utils/README.md

```diff
@@ -0,0 +1,406 @@
+  "p4info": "build/basic.p4.p4info.txt",
+  "bmv2_json": "build/basic.json",
+  "table_entries": [
+    {
+      "table": "MyIngress.ipv4_lpm",
+      "match": {
+        "hdr.ipv4.dstAddr": ["10.0.1.1", 32]
+      },
+      "action_name": "MyIngress.ipv4_forward",
+      "action_params": {
+        "dstAddr": "08:00:00:00:01:11",
+        "port": 1
+      }
+    }
+  ]
+}
+```
+
+### Topology File
+The topology file uses JSON format and contains information regarding the network and its composition. For a simple triangle topology (as used in the tutorials), the topology file takes the format depicted below
+
+![alt text](./img/TopoExample.png "Topology Example")
+ 
+```json
+{
+"hosts": {
+"h1": {"ip": "10.0.1.1/24", "mac": "08:00:00:00:01:11",
+       "commands":["route add default gw 10.0.1.10 dev eth0",
+                   "arp -i eth0 -s 10.0.1.10 08:00:00:00:01:00"]},
+"h2": {"ip": "10.0.2.2/24", "mac": "08:00:00:00:02:22",
+       "commands":["route add default gw 10.0.2.20 dev eth0",
+                   "arp -i eth0 -s 10.0.2.20 08:00:00:00:02:00"]},
+"h3": {"ip": "10.0.3.3/24", "mac": "08:00:00:00:03:33",
+       "commands":["route add default gw 10.0.3.30 dev eth0",
+                   "arp -i eth0 -s 10.0.3.30 08:00:00:00:03:00"]}
+},
+"switches": {
+    "s1": { "runtime_json" : "triangle-topo/s1-runtime.json" },
+    "s2": { "runtime_json" : "triangle-topo/s2-runtime.json" },
+    "s3": { "runtime_json" : "triangle-topo/s3-runtime.json" }
+},
+"links": [
+    ["h1", "s1-p1"], ["s1-p2", "s2-p2"], ["s1-p3", "s3-p2"],
+    ["s3-p3", "s2-p3"], ["h2", "s2-p1"], ["h3", "s3-p1"]
+]
+}
+```
+
+In this file, the following elements are defined:
+
+* **Hosts**. Defines the hosts instantiated by Mininet. Parameters such as IP, MAC address and, startup commands are also defined in this section. In the example, the commands used are:
+
+    * *route add*, which adds a static route to the default gateway (the switch).
+    * *arp*, which manipulates the sytem ARP scache (adds an entry for the switch's MAC address).
+* **Switches**. Defines switch behavior. Three parameters can be inserted here:
+    
+    * *program*. Defines the program inserted into the switch (data plane). <u>Note</u>: If this parameter is not set, then the execution assumes the default P4 file (passed on startup).
+    * *runtime_json*. Defines the path for the control plane file.
+    * *runtime_cli*. Also defines the path for the control plane file. This file is directed at actions that are only supported by the switch_cli interface (such as setting up mirroring).
+    
+* **links**. Defines the links between network nodes. The following list format is used: 
+    ```[Node1, Node2, Latency, Bandwidth]```, where nodes can be defined as ```<Hostname>```, for hosts, and ```<SwitchName>-<SwitchPort>``` for switches. Both *latency* and *bandwidth* are optional, where *latency* is an integer defined in milliseconds(ms) and *bandwidth* is a float defined in megabytes.
```

Comment:
I do not know what the units of bandwidth are, but it cannot be megabytes.  It might be megabytes per second, or perhaps it is megabits per second?

Reply:
You are correct, after doing a small research online I can confirm that Mininet uses megabits per second (Mb/s).